### PR TITLE
feat: Gateway supports the configuration of forwarding tcp/udp traffic.

### PR DESCRIPTION
**Pull request by hongzhouzi**
Signed-off-by: hongzhouzi <hongzhouzi@kubesphere.io>

### What type of PR is this?
/kind feature

### What this PR does / why we need it:
Gateway supports the configuration of forwarding tcp/udp traffic.

### Which issue(s) this PR fixes:

Fixes # https://github.com/kubesphere/kubesphere/issues/5406

### Special notes for reviewers:
Add tcp/udp two configuration items below the configuration options on the gateway edit page, and add tcp/udp configuration in the following format when create and update a **Gateway CR**.

**Configure key-value pairs in tcp/udp configuration, the key is the external port to use and the value indicates the service to expose using the format: `<namespace/service name>:<service port>:[PROXY]:[PROXY]`**
Reference: https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services/

```
apiVersion: gateway.kubesphere.io/v1alpha1
kind: Gateway
metadata: {}
spec:
  controller:
    tcp:
      '8080': 'default/example-tcp-svc1:9000'
      '8081': 'default/example-tcp-svc2:9001'
    udp:
      '53': 'kube-system/coredns:53'
```

### Does this PR introduced a user-facing change?

```release-note
Gateway supports the configuration of forwarding tcp/udp traffic.
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/kubesphere.io/api/gateway/v1alpha1/gateway_types.go [hunk 1]
```diff
@@ -38,6 +38,10 @@ type ControllerSpec struct {
 	Config map[string]string `json:"config,omitempty"`
 	// +optional
 	Scope Scope `json:"scope,omitempty"`
+	// +optional
+	Tcp map[string]string `json:"tcp,omitempty"`
```
Comment (iawia002) [inline staging/src/kubesphere.io/api/gateway/v1alpha1/gateway_types.go:42]: I prefer `TCP`

File: staging/src/kubesphere.io/api/gateway/v1alpha1/gateway_types.go [hunk 2]
```diff
@@ -38,6 +38,10 @@ type ControllerSpec struct {
...
+	// +optional
+	Udp map[string]string `json:"udp,omitempty"`
```
Comment (iawia002) [inline staging/src/kubesphere.io/api/gateway/v1alpha1/gateway_types.go:44]:
```suggestion
	UDP map[string]string `json:"udp,omitempty"`
```

**Comment by hongzhouzi** [comment]
/cc @harrisonliu5 
Can you add this part of the operation and the parameters passed on the front-end page? Both the cluster gateway and the project gateway need to be added.


> Add tcp/udp two configuration items below the configuration options on the gateway edit page, and add tcp/udp configuration in the following format when create and update a **Gateway CR**.
> 
> **Configure key-value pairs in tcp/udp configuration, the key is the external port to use and the value indicates the service to expose using the format: `<namespace/service name>:<service port>:[PROXY]:[PROXY]`** Reference: https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services/
> 
> ```
> apiVersion: gateway.kubesphere.io/v1alpha1
> kind: Gateway
> metadata: {}
> spec:
>   controller:
>     tcp:
>       '8080': 'default/example-tcp-svc1:9000'
>       '8081': 'default/example-tcp-svc2:9001'
>     udp:
>       '53': 'kube-system/coredns:53'
> ```

**Pull request merged**